src/main/java/org/reaktivity/specification/amqp/internal/AmqpFunctions.java

```diff
@@ -0,0 +1,391 @@
+/**
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.specification.amqp.internal;
+
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.kaazing.k3po.lang.el.Function;
+import org.kaazing.k3po.lang.el.spi.FunctionMapperSpi;
+import org.reaktivity.specification.amqp.internal.types.*;
+import org.reaktivity.specification.amqp.internal.types.control.AmqpRouteExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpAbortExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpBeginExFW;
+import org.reaktivity.specification.amqp.internal.types.stream.AmqpDataExFW;
+
+import java.nio.charset.StandardCharsets;
+import java.util.function.Consumer;
+
+public final class AmqpFunctions
+{
+    private static final int MAX_BUFFER_SIZE = 1024 * 8;
+
+    public static class AmqpRouteExHelper
+    {
+        private final AmqpRouteExFW.Builder amqpRouteExRW;
+
+        public AmqpRouteExHelper()
+        {
+            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
+            this.amqpRouteExRW = new AmqpRouteExFW.Builder()
+                .wrap(writeBuffer, 0, writeBuffer.capacity());
+        }
+
+        public AmqpRouteExHelper targetAddress(
+            String targetAddress)
+        {
+            amqpRouteExRW.targetAddress(targetAddress);
+            return this;
+        }
+
+        public AmqpRouteExHelper role(
+            String role)
+        {
+            amqpRouteExRW.role(
+                roleRW -> roleRW.set(AmqpRole.valueOf(role))
+            );
+            return this;
+        }
+
+        public byte[] build()
+        {
+            final AmqpRouteExFW amqpRouteEx = amqpRouteExRW.build();
+            final byte[] result = new byte[amqpRouteEx.sizeof()];
+            amqpRouteEx.buffer().getBytes(0, result);
+            return result;
+        }
+    }
+
+    public static class AmqpBeginExHelper
```

Comment:
Rename `AmqpBeginExHelper` to `AmqpBeginExBuilder` as it is really a builder and has a `build()` method.
Please apply same feedback for other "helper" classes.